### PR TITLE
Emit event when controller delete pod

### LINKED ISSUE #17826: Emit event on scaling replication controller
Currently RC emits an event when it's creating a pod:
https://github.com/kubernetes/kubernetes/blob/d42030170b2b20cb61cb79dff7d095998b7ee7b1/pkg/controller/controller_utils.go#L310

But nothing is emitted when a pod is deleted. We should create event for scaling both ways.

cc @mikedanese @davidopp @fgrzadkowski @smarterclayton

### PR DESCRIPTION
**Pull request by jiangyaoguo**
fixed #17826

### CODE REVIEW

**Comment by gmarek** [comment]
Thanks @jiangyaoguo for taking this!

**Comment by jiangyaoguo** [comment]
@gmarek @eparis Need review.

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -223,8 +223,8 @@ type PodControlInterface interface {
 	CreatePods(namespace string, template *api.PodTemplateSpec, object runtime.Object) error
 	// CreatePodsOnNode creates a new pod accorting to the spec on the specified node.
 	CreatePodsOnNode(nodeName, namespace string, template *api.PodTemplateSpec, object runtime.Object) error
-	// DeletePod deletes the pod identified by podID.
-	DeletePod(namespace string, podID string) error
+	// DeletePod deletes the pod controlled by controller
+	DeletePod(namespace string, pod *api.Pod, object runtime.Object) error
```
Comment (gmarek) [inline pkg/controller/controller_utils.go]: Why did you change from podID (pod.Name) to full pod?. You're using only pod.Name in the implementation.

File: pkg/controller/daemon/controller.go [hunk 2]
```diff
@@ -352,7 +352,8 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) {
 	if err != nil {
 		glog.Errorf("Couldn't get list of nodes when syncing daemon set %+v: %v", ds, err)
 	}
-	var nodesNeedingDaemonPods, podsToDelete []string
+	var nodesNeedingDaemonPods []string
+	var podsToDelete []*api.Pod
```
Comment (gmarek) [inline pkg/controller/daemon/controller.go]: If you pass string instead of *pod this change is not needed.

File: pkg/controller/controller_utils.go (hunk 1 above)
Comment (jiangyaoguo) [inline pkg/controller/controller_utils.go]: You're right. We need podID here only. Probably I considered recording more information of pod once and forgot reverting it.  Will revert it soon. Thanks for your carefully review.

File: pkg/controller/controller_utils.go [hunk 3]
```diff
@@ -230,8 +230,8 @@ type PodControlInterface interface {
 	CreatePods(namespace string, template *api.PodTemplateSpec, object runtime.Object) error
 	// CreatePodsOnNode creates a new pod accorting to the spec on the specified node.
 	CreatePodsOnNode(nodeName, namespace string, template *api.PodTemplateSpec, object runtime.Object) error
-	// DeletePod deletes the pod identified by podID.
-	DeletePod(namespace string, podID string) error
+	// DeletePod deletes the pod controlled by controller
```
Comment (gmarek) [inline pkg/controller/controller_utils.go]: Please revert comment change.

**Comment by gmarek** [comment]
One nit, but otherwise LGTM.

**Comment by jiangyaoguo** [comment]
@gmarek Fixed nit.

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@jiangyaoguo - can you please rebase?

00:28:38 +++ [1220 08:25:56] linux/amd64: go build started
00:28:38 # k8s.io/kubernetes/pkg/controller/job
00:28:38 _output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/controller.go:349: not enough arguments in call to jm.podControl.DeletePod

**Comment by jiangyaoguo** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@jiangyaoguo - your code doesn't compile:

09:43:11 +++ [1220 17:42:39] linux/amd64: go build started
09:43:11 # k8s.io/kubernetes/pkg/controller/job
09:43:11 _output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/job/controller.go:349: not enough arguments in call to jm.podControl.DeletePod

**Comment by jiangyaoguo** [comment]
@k8s-bot test this please

**Comment by gmarek** [comment]
@k8s-bot unit test this

**Comment by gmarek** [comment]
@wojtek-t - can you take a look at the unit test failure and triage if this is go1.5 issue?

**Comment by wojtek-t** [comment]
@gmarek - this is https://github.com/kubernetes/kubernetes/issues/18928 (I hope the PR fixing it is already in-flight)

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by jiangyaoguo** [comment]
@wojtek-t @gmarek Thanks for your help.
